src/components/Signin/Signin.jsx

```diff
@@ -1,36 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
+
 import { useStore } from "../../store";
+import { useFormWithValidation } from "../../utils/helpHooks";
 
 export default function Signin() {
+    const [error, setError] = useState("");
+    const { signinUser } = useStore().currentUser;
     const { setRegisteredFalse } = useStore().userRegistration;
+    const { closePopup } = useStore().popupWithForm;
+    const { values, handleChange, errors, isValid } = useFormWithValidation();
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try {
+            const res = await signinUser(values);
+            if (res._id) {
+                closePopup();
+                window.location.reload();
+            } else {
+                setError(res.message);
+                setTimeout(() => {
+                    setError("");
+                }, 2000);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+    };
 
     return (
         <>
             <h2 className='popup__title'>Sign in</h2>
-            <form className='popup__form'>
+            <form className='popup__form' onSubmit={handleSubmit}>
                 <div className='popup__form-labels'>
                     <label className='popup__form-label'>
                         Email:
                         <input
+                            name='email'
                             className='popup__form-input'
                             type='email'
                             placeholder='Enter email'
+                            onChange={handleChange}
                             required
                         />
-                        <p className='popup__form-input_type_error'></p>
+                        <p className='popup__form-input_type_error'>
+                            {errors.email && "Invalid email address"}
+                        </p>
                     </label>
                     <label className='popup__form-label'>
                         Password:
                         <input
+                            name='password'
                             placeholder='Enter password'
                             className='popup__form-input'
                             type='password'
+                            onChange={handleChange}
```

Comment:
If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about **controlled components**: https://reactjs.org/docs/forms.html#controlled-components